pkg/common/patterns.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const EmailPattern = `\b([a-z0-9]{4,25}@[a-zA-Z0-9]{2,12}.[a-zA-Z0-9]{2,6})\b`
+const SubDomainPattern = `\b([0-9a-zA-Z]{2,40})\b`
+const UUIDPattern = `\b([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\b`
+const UUIDPatternUpperCase = `\b([0-9A-Z]{8}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{12})\b`
+
+const RegexNumPattern = "0-9"
+const RegexPattern = "0-9a-z"
+const RegexPatternUpperCase = "0-9A-Z"
+const Base64Pattern = "0-9a-zA-Z"
+const HexPattern = "0-9a-f"
+const HexPatternUpperCase = "0-9A-F"
+const HexPatternWithUpperCase = "0-9a-fA-F"
+
+//Custom Regex functions
+func BuildRegex(pattern string, length int) string {
+	return fmt.Sprintf(`\b([%s]{%s})\b`, pattern, strconv.Itoa(length))
+}
+
+func BuildRegexWithChar(pattern, specialChar string, length int) string {
+	return fmt.Sprintf(`\b([%s%s]{%s})\b`, pattern, specialChar, strconv.Itoa(length))
+}
+
+func BuildRegexJWT(firstRange, secondRange, thirdRange string) string {
+	first_range_split := strings.Split(firstRange, ",")
+	first_range_min := strings.TrimSpace(first_range_split[0])
+	first_range_max := strings.TrimSpace(first_range_split[1])
+	if first_range_min >= first_range_max {
+		log.Error("First min value should not be greater than or equal to max")
```

Comment:
These should be a panic, which will be caught during the test. Also, does numerical comparison work on strings?

Reply:
Based on what I've read here `https://stackoverflow.com/questions/31487694/how-can-i-convert-string-to-integer-in-golang`, as long as it doesn't need the fractional part, then it will work. I tested it in playground as well `https://go.dev/play/p/Oc-_X_cFzrd`

Reply:
The string comparison fails with "100,22" and "33,222"

Reply:
Oh, thanks for pointing this out! I just pushed the changes and used parseInt()